proposals/lockout-violation-detection.md

```diff
@@ -1,56 +1,96 @@
 ### Lockout Violation Detection
 
-We assume a database that we will store vote information ingested from
+Assume:
+1. A database that we will store vote information ingested from
 gossip/turbine, which in turn will be consulted to detect lockout violations
 as follows.
+2. First we track all SlotHashes on the canonical/rooted fork in the database
+for the last epoch.
 
-1) Track all SlotHashes on the main fork in the database.
-2) If there are any votes made that have a root `R` missing from the SlotHashes
-list, slash them on this fork
-3) For all other slots `S` in each vote other than the root, track a range
+The following sections will go over the possible lockout violations that will
+be detected and punished.
+
+#### Rooting a different fork
+If there are any votes made that have a root `R` missing from the rooted
+SlotHashes list, slash them on the rooted fork since they have committed
+to a different fork
+
+#### Removing lockouts
+Attempting to illegally *remove* a lockout in a newer vote that that should
+have still existed based on an older vote is a lockout violation. We detect
+this as follows:
+
+1. For all other slots `S` in each vote other than the root, track a range
```

Comment:
nit: "For all non root slots `S` in each vote" sounds cleaner without the double "other"

Reply:
fixed